### PR TITLE
Vendor dependencies with govendor

### PR DESCRIPTION
**Pull request by metcalf**
Neither of the two existing Go dependencies discuss versioning or API compatibility. In https://github.com/burke/zeus/pull/550 I'm proposing to introduce a dependency on an external library that [explicitly anticipates API changes](https://github.com/fsnotify/fsevents/commit/83a642eb3ec3336b04a0cfaff0a897dc26652790#diff-04c6e90faac2675aa89e2176d2eec7d8).

Vendoring dependencies into zeus breaks our reliance on the stability of these libraries and ensures reproducible builds. Vendor support in the Go toolchain is now standard in Go 1.6 and [recommended for vendoring dependencies by the Go team](https://github.com/golang/go/wiki/PackageManagementTools#go15vendorexperiment).

### CODE REVIEW

**Comment by metcalf** [comment]
@latortuga would love to get your take on this since it's a meaningful change to how zeus is built.

File: Makefile [hunk 1]
```diff
@@ -155,7 +155,6 @@ clean:
 
 .PHONY: dev_bootstrap
 dev_bootstrap: go/zeusversion/zeusversion.go
-	go get ./...
 	bundle -v || gem install bundler --no-rdoc --no-ri
 	bundle install
-	go get github.com/mitchellh/gox
+	go get github.com/mitchellh/gox github.com/kardianos/govendor
```
Comment (antifuchs) [inline Makefile]: I believe this is incorrect - you do need `gox` to cross-compile.

Comment (metcalf) [inline Makefile]: Still there!

**Comment by metcalf** [comment]
Going to go ahead and land this since I've discussed it with @antifuchs and I'm waiting on it for another PR. Happy to discuss further if you have concerns!

**Comment by latortuga** [comment]
Anything to make the build process more reliable and consistent is fine with me. I've had issues in the past where I've tried to build after some time has passed and the golang dependencies have changed. I'd characterize myself as a novice golang programmer so the best practices here are really not something I know much about.

TLDR: :shipit:
